hw/ip/tlul/rtl/tlul_adapter_reg_racl.sv

```
@@ -94,30 +69,62 @@ module tlul_adapter_reg_racl
       .out_o( racl_role_vec )
     );
 
+    logic rd_req, racl_read_allowed, racl_write_allowed;
+    assign rd_req             = tl_i.a_opcode == tlul_pkg::Get;
     assign racl_read_allowed  = (|(racl_policies_i[RaclPolicySelVec].read_perm  & racl_role_vec));
     assign racl_write_allowed = (|(racl_policies_i[RaclPolicySelVec].write_perm & racl_role_vec));
-    assign racl_error_o       = (rd_req & ~racl_read_allowed) | (wr_req & ~racl_write_allowed);
-    // RACL only generates error responeses if enabled
-    assign racl_error         = racl_error_o & RaclErrorRsp;
+    assign racl_error_o       = (rd_req & ~racl_read_allowed) | (~rd_req & ~racl_write_allowed);
+
+    tlul_request_loopback #(
+      .ErrorRsp(RaclErrorRsp)
```

Do we ever expect to instantiate this loopback module without setting `ErrorRsp`? I'm struggling to think of a use case for that. Maybe we should drop the parameter?

---

Yes, we defined RACL to support 2 forms of error behavior:
1. Return a TLUL error on a RACL block
2. Silently drop a write and return all 1 on read
For case number 2, we need the parametrized error response on the loopback.

---

Ahah, that makes sense. Thanks for explaining.